### PR TITLE
Make sure `MeshLibrary` shape array has correct number of elements

### LINKED ISSUE #53897: Can't add element to the MeshLibrary Item Shapes array
### Godot version

3.3.stable

### System information

Archlinux

### Issue description

When trying to increase number of elements of the `MeshLibrary/Item[]/Shapes` array Godot fails with:
```scene/resources/mesh_library.cpp:262 - Condition "p_shapes.size() & 1" is true.```

image

### Steps to reproduce

1. Create an empty `MeshLibrary`,
2. Add an item,
3. Increase the `Shapes` array size.

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by rafallus**
The shape array shown in the inspector increases and decreases by 2, ensuring that the last elements (when increasing) is a shape-transform pair.
Fixes: #53897

### CODE REVIEW

File: scene/resources/mesh_library.cpp [hunk 1]
```diff
@@ -29,6 +29,7 @@
 /*************************************************************************/
 
 #include "mesh_library.h"
+#include "box_shape_3d.h"
```
Comment (akien-mga) [inline scene/resources/mesh_library.cpp:32]: Nitpick but to match our preferred style from https://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html#header-includes
```suggestion

#include "box_shape_3d.h"
```

File: scene/resources/mesh_library.cpp [hunk 2]
```diff
@@ -255,12 +256,35 @@ int MeshLibrary::get_last_unused_item_id() const {
...
+		if (prev_size < size) {
+			// Check if last element is a shape
+			Ref<Shape3D> shape = arr_shapes[size - 1];
+			if (shape.is_null()) {
+				Ref<BoxShape3D> box_shape;
+				box_shape.instantiate();
+				arr_shapes[size - 1] = box_shape;
+			}
+
+			// Make sure the added element is a Transform3D
```
Comment (akien-mga) [inline scene/resources/mesh_library.cpp:274]:
```suggestion
			// Check if last element is a shape.
			Ref<Shape3D> shape = arr_shapes[size - 1];
			if (shape.is_null()) {
				Ref<BoxShape3D> box_shape;
				box_shape.instantiate();
				arr_shapes[size - 1] = box_shape;
			}

			// Make sure the added element is a Transform3D.
```

File: scene/resources/mesh_library.cpp [hunk 3]
```diff
@@ -255,12 +256,35 @@ int MeshLibrary::get_last_unused_item_id() const {
...
+		ERR_FAIL_COND_MSG(!item_map.has(p_item), "Requested for nonexistent MeshLibrary item '" + itos(p_item) + "'.");
+		int prev_size = item_map[p_item].shapes.size() * 2;
+
+		if (prev_size < size) {
+			// Check if last element is a shape
+			Ref<Shape3D> shape = arr_shapes[size - 1];
+			if (shape.is_null()) {
+				Ref<BoxShape3D> box_shape;
+				box_shape.instantiate();
+				arr_shapes[size - 1] = box_shape;
```
Comment (akien-mga) [inline scene/resources/mesh_library.cpp:271]: I'm not sure about this, that seems pretty arbitrary, and could lead to a lot of instantiations that will then need to be freed and replaced by what the user wants to use.

Wouldn't it be sufficient to just make the step 2 for the SpinBox?

Comment (akien-mga) [reply scene/resources/mesh_library.cpp:271]: > Wouldn't it be sufficient to just make the step 2 for the SpinBox?

Well not sure that's actually possible currently in the array editor.

Comment (rafallus) [reply scene/resources/mesh_library.cpp:271]: The array is expecting to have a shape. If there is anything else, even an empty `Variant()` the array won't update and the size won't change. A few lines below that piece of code you can find
```gdscript
if (sd.shape.is_valid()) {
	shapes.push_back(sd);
}
```
So, if it isn't a valid shape, nothing happens. Could you think of other way of doing it? It is absolutely arbitrary but is the way I found to do it.

File: scene/resources/mesh_library.cpp (hunk 1 above)
Comment (rafallus) [reply scene/resources/mesh_library.cpp:33]: Done

File: scene/resources/mesh_library.cpp (hunk 2 above)
Comment (rafallus) [reply scene/resources/mesh_library.cpp]: And done

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.5.

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.3.

**Pull request merged**